code/modules/plumbing/plumbers/plumbing/fountain.dm

```diff
@@ -0,0 +1,63 @@
+/obj/machinery/plumbing/fountain
+	name = "fountain"
+	desc = "A fountain, you can drink from here."
+	icon_state = "fountain"
+	density = TRUE
+	capacity = 200
+	deployable = /obj/item/deployable/fountain
+	///fluid overlay that appears over the fountain when there are liquids inside, it changes color
+	var/mutable_appearance/fluid_overlay
+	///if it has the overlay already
+	var/has_fluid = FALSE
+
+/obj/machinery/plumbing/fountain/Initialize()
+	. = ..()
+	create_reagents(capacity, DRAINABLE|AMOUNT_VISIBLE)
+	AddComponent(/datum/component/plumbing/output)
+
+	fluid_overlay = fluid_overlay || mutable_appearance('icons/obj/plumbing/plumbers.dmi')
+	fluid_overlay.icon_state = "fountain_grey"
+	fluid_overlay.plane = BELOW_MOB_LAYER
+
+/obj/machinery/plumbing/fountain/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	..()
+
+/obj/machinery/plumbing/fountain/process()
+	update_icon()
```

Comment:
It would be a lot more sane if you just called update_icon when there's a change in reagents, power or anchored status, instead of every process.

Reply:
is there a check for reagent changes?

Reply:
on_reagent_change